stdlib/public/core/UnsafeRawBufferPointer.swift.gyb

```diff
@@ -644,21 +652,26 @@ extension Unsafe${Mutable}RawBufferPointer {
       return (it, UnsafeMutableBufferPointer(start: nil, count: 0))
     }  
 
+    _debugPrecondition(_end != nil)
+    var newCount = 0
     for p in stride(from: base, 
       // only advance to as far as the last element that will fit
-      to: base + count - elementStride + 1, 
+      // We unsafely unwrap because _end cannot be nil if _base is not.
+      to: _end._unsafelyUnwrappedUnchecked - elementStride + 1, 
       by: elementStride
     ) {
       // underflow is permitted -- e.g. a sequence into
       // the spare capacity of an Array buffer
       guard let x = it.next() else { break }
       p.initializeMemory(as: S.Element.self, repeating: x, count: 1)
       formIndex(&idx, offsetBy: elementStride)
+      // Unchecked, cannot overflow.
+      newCount &+= 1
     }
 
     return (it, UnsafeMutableBufferPointer(
                   start: base.assumingMemoryBound(to: S.Element.self), 
-                  count: idx / elementStride))
+                  count: newCount))
```

Comment:
Are you sure this transformation is actually worth it?

Reply:
> Good question, I made this transformation on gut-feel so I'm definitely not sure. My gut instinct here is that the additions are _approximately_ free on all modern hardware as the only data dependency it has is on the previous loop iteration and most modern chips are wide enough to issue that instruction in parallel with the rest. That makes it strictly cheaper than a checked division.

That matches my intuition. I've always jumped through hoops to avoid integer division, although admittedly it's much faster than it used to be.

Reply:
I don't think we need to overly worry about integer divisions by constants these days. If this PR is about code size, then we should go with whichever variant is shorter.

Reply:
I'm never worried about integer division by a constant, especially power of two, because it doesn't generate any integer division. Sorry for that confusion.

Reply:
FWIW, this code is not eliding the integer division: in either case, it produces `sdiv`.

Reply:
However, to be clear, I think it's not worth litigating this on this PR. I'll do an investigation separately once the other uncontroversial changes land.